control/server.py

```diff
@@ -41,6 +42,17 @@ def callable():
 
     return callable
 
+def wait_for_file(file_path, timeout=None, interval=1):
+    start_time = time.time()
+
+    while True:
+        if os.path.exists(file_path):
+            return
+
+        if timeout is not None and time.time() - start_time >= timeout:
+            raise Exception(f"Timeout: file '{file_path}' not found after {timeout} seconds.")
```

Comment:
```suggestion
        if timeout is not None and (elapsed_time := time.time() - start_time) >= timeout:
            raise Exception(f"Timeout: file '{file_path}' not found after {elapsed_time} seconds.")
```

Reply:
@epuertat 🖖, not sure the suggested syntax works, since, In Python, the assignment statement itself does not return a value.  Python treats an assignment as a statement rather than an expression:
```python
Python 3.7.9 (v3.7.9:13c94747c7, Aug 15 2020, 01:31:08)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import time
>>> start_time = time.time()
>>> if (elapsed := time.time() - start_time) >= 0:
  File "<stdin>", line 1
    if (elapsed := time.time() - start_time) >= 0:
                ^
SyntaxError: invalid syntax
>>> if (elapsed = time.time() - start_time) >= 0:
  File "<stdin>", line 1
    if (elapsed = time.time() - start_time) >= 0:
                ^
SyntaxError: invalid syntax
>>>
```

Reply:
It works in Python >=3.8 (btw Python 3.7 EoL ended a couple of months ago).